examples/ensemble/plot_isolation_forest.py

```diff
@@ -1,67 +1,114 @@
+
+# %%
+# We can visualize the resulting clusters:
+
 import matplotlib.pyplot as plt
-from sklearn.ensemble import IsolationForest
 
-rng = np.random.RandomState(42)
+plt.scatter(X[:, 0], X[:, 1], c=y, s=20, edgecolor="k")
+plt.title("Gaussian Mixture clusters")
+plt.show()
+
+# %%
+# We train the model and use the class
+# :class:`~sklearn.inspection.DecisionBoundaryDisplay` to visualize a discrete
+# decision boundary to determine whether a particular sample is an outlier or
+# not.
 
-# Generate train data
-X = 0.3 * rng.randn(100, 2)
-X_train = np.r_[X + 2, X - 2]
-# Generate some regular novel observations
-X = 0.3 * rng.randn(20, 2)
-X_test = np.r_[X + 2, X - 2]
-# Generate some abnormal novel observations
-X_outliers = rng.uniform(low=-4, high=4, size=(20, 2))
+import matplotlib.pyplot as plt
+from sklearn.inspection import DecisionBoundaryDisplay
+from sklearn.ensemble import IsolationForest
 
-# fit the model
 clf = IsolationForest(max_samples=100, random_state=rng)
 clf.fit(X_train)
-y_pred_train = clf.predict(X_train)
-y_pred_test = clf.predict(X_test)
-y_pred_outliers = clf.predict(X_outliers)
-
-# plot the line, the samples, and the nearest vectors to the plane
-xx, yy = np.meshgrid(np.linspace(-5, 5, 50), np.linspace(-5, 5, 50))
-Z = clf.decision_function(np.c_[xx.ravel(), yy.ravel()])
-Z = Z.reshape(xx.shape)
-
-plt.title("IsolationForest")
-plt.contourf(xx, yy, Z, cmap=plt.cm.Blues_r)
-
-b1 = plt.scatter(X_train[:, 0], X_train[:, 1], c="white", s=20, edgecolor="k")
-b2 = plt.scatter(X_test[:, 0], X_test[:, 1], c="green", s=20, edgecolor="k")
-c = plt.scatter(X_outliers[:, 0], X_outliers[:, 1], c="red", s=20, edgecolor="k")
-plt.axis("tight")
-plt.xlim((-5, 5))
-plt.ylim((-5, 5))
-plt.legend(
-    [b1, b2, c],
-    ["training observations", "new regular observations", "new abnormal observations"],
+
+disp = DecisionBoundaryDisplay.from_estimator(
+    clf,
+    X,
+    response_method="predict",
+    alpha=0.5,
+)
+scatter = disp.ax_.scatter(X[:, 0], X[:, 1], c=y, s=20, edgecolor="k")
+disp.ax_.legend(
+    *scatter.legend_elements(),
+    title="True class",
+    loc="upper left",
+)
+disp.ax_.set_title("Binary decision boundary of IsolationForest")
+plt.show()
+
+# %%
+# By setting the `response_method="decision_function"`, the
+# :class:`~sklearn.inspection.DecisionBoundaryDisplay` plots instead the measure
+# of normality of an observation, which is given by the depth of the leaf (or
+# equivalently the number of splits) required to isolate a sample in a given
+# position.
+
+disp = DecisionBoundaryDisplay.from_estimator(
+    clf,
+    X,
+    response_method="decision_function",
+    alpha=0.5,
+)
+scatter = disp.ax_.scatter(X[:, 0], X[:, 1], c=y, s=20, edgecolor="k")
+disp.ax_.legend(
+    *scatter.legend_elements(),
+    title="True class",
     loc="upper left",
 )
+disp.ax_.set_title("Path length decision boundary of IsolationForest")
```

Comment:
It could be nice to have a colorbar or have the level of the isoline in terms of the decision. I don't recall if setting `plot_method="pcolormesh"` would make the level of the isoline. But we can for sure add a colorbar otherwise (cf. [quick example](https://matplotlib.org/stable/gallery/subplots_axes_and_figures/subplots_adjust.html#sphx-glr-gallery-subplots-axes-and-figures-subplots-adjust-py))

Reply:
It's done, but I am having problems with interpreting the colorbar, so any feedback on that side is really appreciated!

Reply:
It corresponds to the score of confidence to be an outlier or not. 1 is an inlier and 0 is an outlier.